lib/config/types.ts

```diff
@@ -134,7 +135,7 @@ export interface LegacyAdminConfig {
   onboardingConfigFileName?: string;
 
   platform?: string;
-  requireConfig?: boolean;
+  requireConfig?: RequiredConfig | boolean;
```

Comment:
why still boolean?
```suggestion
  requireConfig?: RequiredConfig;
```

Reply:
Probably for tests :)

Reply:
yes, this is for tests.
should I remove it and find another way to make the tests pass or just leave it as is?

Reply:
Remove it, because you should not leave incorrect types in production in order to get tests to pass. You can use the TypeScript `as` keyword in tests to bypass the checks where necessary.